storage/tianmu/core/query.cpp

```diff
@@ -1122,33 +1122,27 @@ CondID Query::ConditionNumberFromMultipleEquality(Item_equal *conds, const TabID
   ifield = li++;
   if (!Item2CQTerm(ifield, first_term, tmp_table, filter_type)) return CondID(-1);
   CondID filter;
-  if (!and_me_filter)
-    cq->CreateConds(filter, tmp_table, first_term, common::Operator::O_EQ, zero_term, CQTerm(),
+  cq->CreateConds(filter, tmp_table, first_term, common::Operator::O_EQ, zero_term, CQTerm(),
                     is_or_subtree || filter_type == CondType::HAVING_COND);
-  else {
-    if (is_or_subtree)
-      cq->Or(*and_me_filter, tmp_table, first_term, common::Operator::O_EQ, zero_term);
-    else
-      cq->And(*and_me_filter, tmp_table, first_term, common::Operator::O_EQ, zero_term);
-  }
+
   while ((ifield = li++) != nullptr) {
     if (!Item2CQTerm(ifield, next_term, tmp_table, filter_type)) return CondID(-1);
-    if (!and_me_filter) {
```

Comment:
as said in 
`
        while ((item = li++) != nullptr) {
          CondID res = ConditionNumber(item, tmp_table, filter_type, and_cond.get(),
                                       // if there is no and_cond the filter has to be created and info
                                       // about tree/no tree like descriptor has to be passed recursively
                                       // down once filter is created we pass 'false' to indicate AND
                                       and_cond.get() ? false : is_or_subtree);
`

Be here, `is_or_filter` should be always `false`.

Reply:
Cette fonction traite de l'optimisation équivalente, de l'optimisation équivalente ou de l'optimisation de transfert constant, qui convertit plusieurs conditions en une seule condition, donc utilisez et.